superset/assets/src/explore/components/controls/DateFilterControl.jsx

```diff
@@ -1,20 +1,43 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import {
-  Button, ButtonGroup, FormControl, InputGroup,
-  Label, OverlayTrigger, Popover, Glyphicon,
+  Button,
+  DropdownButton,
+  FormControl,
+  FormGroup,
+  InputGroup,
+  Label,
+  MenuItem,
+  OverlayTrigger,
+  Popover,
+  Radio,
+  Tab,
+  Tabs,
 } from 'react-bootstrap';
-import Select from 'react-select';
-import Datetime from 'react-datetime';
 import 'react-datetime/css/react-datetime.css';
+import DateTimeField from 'react-bootstrap-datetimepicker';
+import 'react-bootstrap-datetimepicker/css/bootstrap-datetimepicker.min.css';
 import moment from 'moment';
 
 import ControlHeader from '../ControlHeader';
-import PopoverSection from '../../../components/PopoverSection';
+import InfoTooltipWithTrigger from '../../../components/InfoTooltipWithTrigger';
+import { t } from '../../../locales';
 
-const RELATIVE_TIME_OPTIONS = ['ago', 'from now'];
+const TYPES = ['range', 'startend'];
+const COMMON_TIME_FRAMES = ['Yesterday', 'Last week', 'Last month', 'Last year'];
+const RELATIVE_TIME_OPTIONS = ['Last', 'Next'];
 const TIME_GRAIN_OPTIONS = ['seconds', 'minutes', 'hours', 'days', 'weeks', 'months', 'years'];
 
+const MOMENT_FORMAT = 'YYYY-MM-DD[T]HH:mm:ss';
+const DEFAULT_SINCE = moment().startOf('day').subtract(7, 'days').format(MOMENT_FORMAT);
+const DEFAULT_UNTIL = moment().startOf('day').format(MOMENT_FORMAT);
+const INVALID_DATE_MESSAGE = 'Invalid date';
+const SEPARATOR = ' : ';
+const FREEFORM_TOOLTIP = t(
+  'Superset supports smart date parsing. Strings like `last sunday` or ' +
```

Comment:
[honest question] is this functionality worth supporting? It seems like if I wasn't a power user I wouldn't know which keywords work and which don't. For example, would "2 weeks ago" work? What about "the quarter before last"

Reply:
Even if we decided we're better off without it, it'd be hard to deprecate (would break charts).

Reply:
Yeah, without feedback informing if the input is valid this doesn't seem very usable. We could do a backend call and show it in red if not supported?